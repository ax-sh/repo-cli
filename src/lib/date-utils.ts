import spacetime from 'spacetime'

export function elapsedTime(date: string) {
  const now = spacetime.now()
  return now.since(spacetime(date)).rounded
}
