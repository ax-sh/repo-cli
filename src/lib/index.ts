import appRootPath from 'app-root-path'

export { appRootPath }

export async function execa() {
  const { execa } = await import('execa')
  return execa
}

export function generateNewCmdPath(name: string) {
  return {
    command: appRootPath.resolve(`src/commands/${name}.ts`),
    service: appRootPath.resolve(`src/services/${name}/${name}.service.ts`),
    test: appRootPath.resolve(`src/services/${name}/${name}.test.ts`),
  }
}

export async function appRootPathLazy() {
  const appRootPathModule = await import('app-root-path')
  return appRootPathModule.default
}

export function enableConsoleDepth() {
  // eslint-disable-next-line no-console
  globalThis.console = new console.Console({
    inspectOptions: {
      depth: null,
    },
    // eslint-disable-next-line node/prefer-global/process
    stdout: process.stdout,
    // eslint-disable-next-line node/prefer-global/process
    stderr: process.stderr,
  })
}
