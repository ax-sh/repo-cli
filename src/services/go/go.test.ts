import { filesystem } from 'gluegun'
import { exeCmdWithOutput } from '../../lib'

vi.mock('../../lib/helpers/cmd/cli')
// @ts-expect-error not a error
vi.mock('gluegun', { filesystem: { isFile: vi.fn() } })

describe('[go] service test', () => {
  it('should check if filesystem has been mocked properly', () => {
    // eslint-disable-next-line ts/unbound-method
    const fn = vi.mocked(filesystem.isFile)

    expect(fn).toHaveProperty('mock')
  })
  it('should addCodeQualityTools', async () => {
    const mod = await import('./go.service')
    expect(mod).toBeDefined()
    // eslint-disable-next-line ts/unbound-method
    vi.mocked(filesystem.isFile).mockReturnValue(true)

    const fn = vi.mocked(exeCmdWithOutput)
    expect(fn).toHaveProperty('mock')
    fn.mockImplementation(async (args: string) => args)
    const out = await mod.addCodeQualityTools()
    expect(fn).toBeCalled()
    expect(fn).toHaveBeenCalledTimes(4)
    expect(fn).toHaveResolved()
    expect(out).toMatchSnapshot()
  })
})
