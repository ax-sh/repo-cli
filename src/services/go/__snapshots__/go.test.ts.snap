// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[go] service test > should addCodeQualityTools 1`] = `
"
  # lint
  go tool golangci-lint run
  # release
  go tool goreleaser --snapshot --clean
  # tidy imports
  go tool goimports -l -w .
  # format
  go tool gofumpt -l -w .
  "
`;
